File: bench/fixtures/sast/javascript/cwe78_cmd_injection.js
// CWE-78 fixture: command injection via child_process.exec.
// Expected to be detected by p/javascript or p/owasp-top-ten rules.

const { exec } = require('child_process');

function runUserCommand(userInput) {
  // CWE-78: untrusted input passed to exec (which uses a shell).
  exec(userInput, (err, stdout) => {
    if (err) throw err;
    return stdout;
  });
}

module.exports = { runUserCommand };

File: bench/fixtures/sast/python/cwe78_cmd_injection.py
"""CWE-78 fixture: command injection via subprocess shell=True.

Expected to be detected by p/python or p/owasp-top-ten semgrep rules.
"""

from __future__ import annotations

import subprocess


def run_user_command(user_input: str) -> None:
    # CWE-78: untrusted input passed to a shell.
    subprocess.run(user_input, shell=True, check=False)  # noqa: S602


def list_dir(path: str) -> str:
    # CWE-78: string concatenation into shell.
    return subprocess.check_output("ls " + path, shell=True).decode()  # noqa: S602
